docs/concepts/monitoring.md

```diff
@@ -0,0 +1,115 @@
+- **ELK Stack (Elasticsearch, Logstash, Kibana)**: Centralized log management for troubleshooting and audit purposes.
+
+---
+
+## 2. Objectives
+The primary objectives of this monitoring setup are:
+- **Real-time Monitoring**: Track the health, performance, and availability of all components, including frontend, backend, and core banking system.
+- **Centralized Logging**: Aggregate logs from all parts of the system to a central location for easier troubleshooting and auditing.
+- **Proactive Alerting**: Set up automated alerts for critical issues, such as high CPU usage, slow API response times, or failed transactions.
+- **Performance Optimization**: Continuously monitor key metrics to optimize performance and detect bottlenecks.
+
+---
+
+## 3. Key Metrics to Monitor
+### a. Frontend
+- **API Response Times**: Monitor the response times of API calls from the frontend to the backend.
+- **Page Load Times**: Track the time it takes for the frontend to render the user interface.
+- **Errors**: Monitor any client-side errors or failed API requests.
+
+### b. Backend
+- **CPU and Memory Usage**: Monitor the CPU and memory consumption of the Spring Boot services to ensure resource utilization is within optimal limits.
+- **API Throughput and Latency**: Track the number of API requests handled, response times, and error rates.
+- **Transaction Rates**: Track the number of transactions processed per minute, including failed transactions.
+
+### c. Core Banking System
+- **Transaction Volumes**: Monitor the volume of transactions processed by the core banking system.
+- **System Uptime**: Ensure that the core banking system is up and running consistently.
+- **Integration Failures**: Monitor any errors or timeouts when integrating with third-party banking services.
+
+### d. Infrastructure
+- **Server Health**: Monitor CPU, memory, and disk usage on all nodes (CDN, API gateway, load balancers, backup servers).
+- **Network Traffic**: Track the amount of network traffic flowing through load balancers and API gateways.
+
+
+---
+
+## 4. Tools and Setup
+
+### a. Prometheus
+- **Purpose**: Prometheus will be used to scrape metrics from the various components of the system, including the backend services, database, and infrastructure.
+- **Integration**: Each service will expose an endpoint (`/metrics`) from which Prometheus can scrape performance data.
+- **Key Metrics**:
+  - API latency and throughput
+  - System resource usage (CPU, memory)
+
+
+### b. Grafana
+- **Purpose**: Grafana will be used to visualize the data collected by Prometheus through custom dashboards.
+- **Dashboards**: Create dashboards for different environments (development, staging, production) to provide real-time visibility into system performance.
+- **Key Visualizations**:
+  - CPU and memory usage over time
+  - API request/response rates
+  - Transaction volumes
+  
+
+### c. ELK Stack (Elasticsearch, Logstash, Kibana)
```

Comment:
The ELK stack is primarily focused on logging and we have a ticket for that so this setup should not be here.

Reply:
Yes yes..

let me correct that now @nancymuyeh